templates/workloads/partials/cf/subscribe.yml

```
@@ -0,0 +1,226 @@
+    {{- if .Subscribe.Queue.Retention}}
+    MessageRetentionPeriod: {{.Subscribe.Queue.Retention}}
+    {{- end}}
+    {{- if .Subscribe.Queue.Delay}}
+    DelaySeconds: {{.Subscribe.Queue.Delay}}
+    {{- end}}
+    {{- if .Subscribe.Queue.Timeout}}
+    VisibilityTimeout: {{.Subscribe.Queue.Timeout}}
+    {{- end}}
+    {{- if .Subscribe.Queue.KMS}}
+    KmsMasterKeyId: !Ref SQSQueueKMSKey
+    {{- else}}
+    KmsMasterKeyId: 'alias/aws/sqs'
+    {{- end}}
+    {{- if .Subscribe.Queue.FIFO}}
+    FifoQueue: true
+    {{- if .Subscribe.Queue.FIFO.HighThroughput}}
+    FifoThroughputLimit: 'perMessageGroupId'
+    DeduplicationScope: 'messageGroup'
+    {{- end}}
+    {{- end}}
+    {{- if .Subscribe.Queue.DeadLetter}}
+    RedrivePolicy:
+      {{- if .Subscribe.Queue.DeadLetter.Id}}
+      deadLetterTargetArn: !Sub 'arn:${AWS::Partition}:sqs:${AWS::Region}:${AWS::AccountId}:{{logicalIDSafe .Subscribe.Queue.DeadLetter.Id}}'
+      {{- else}}
+      deadLetterTargetArn: !GetAtt DeadLetter.Arn
+      {{- end}}
+      maxReceiveCount: {{.Subscribe.Queue.DeadLetter.Tries}}
+    {{- end}}
+  {{- end}}
+
+{{- if .Subscribe.Queue}}{{- if .Subscribe.Queue.DeadLetter}}{{- if not .Subscribe.Queue.DeadLetter.Id}}
+DeadLetter:
+  Type: AWS::SQS::Queue
+  Properties:
+    QueueName: !Join ['', [!GetAtt SQSQueue.Name, "-", "DeadLetter"]]
+
+DeadLetterPolicy:
+  Type: AWS::SQS::QueuePolicy
+  Properties:
+    Queues: [!Ref 'DeadLetter']
+    PolicyDocument:
+      Version: "2012-10-17"
+      Statement:
+        - Sid: Allow-TaskRole-Subscribe
+          Effect: Allow
+          Principal:
+            Service: sns.amazonaws.com
+          Action: 
+            - sqs:GetMessage
+            - sqs:DeleteMessage
+          Resource: !GetAtt SQSQueue.Arn
+          Condition:
+            ArnEquals:
+              aws:SourceArn: !Ref 'TaskRole'
+{{- end}}{{- end}}{{- end}}
+
+{{- if .Subscribe.Queue}}{{- if .Subscribe.Queue.KMS}}
+SQSQueueKMSKey:
+  Type: AWS::KMS::Key
+  Properties:
+    KeyPolicy:
+      Version: '2012-10-17'
+      Statement:
+        - Effect: Allow
+          Principal:
+            AWS: !Sub 'arn:${AWS::Partition}:iam::${AWS::AccountId}:root'
+          Action: kms:*
+          Resource: '*'
+{{- end}}{{- end}}
+
+QueuePolicy:
+  Type: AWS::SQS::QueuePolicy
+  Properties:
+    Queues: [!Ref 'SQSQueue']
+    PolicyDocument:
+      Version: '2012-10-17'
+      Statement:
+        - Sid: Allow-TaskRole-Subscribe
+          Effect: Allow
+          Principal:
+            Service: sns.amazonaws.com
+          Action: 
+            - sqs:GetMessage
+            - sqs:DeleteMessage
+          Resource: !GetAtt SQSQueue.Arn
+          Condition:
+            ArnEquals:
+              aws:SourceArn: !Ref 'TaskRole'
+        {{- range $topic := .Subscribe.Topics}}
```

I think `range` has to be: 
```suggestion
        {{- range $idx, $topic := .Subscribe.Topics}}
```

---

Does this need to be addressed?

---

Range can be done three ways:
```
{{range .A}}{{.}} {{end}}                       // Simplest possible
{{range $a := .A}}{{$a}} {{end}}                // With one variable
{{range $idx, $a := .A}}{{$a}},{{$idx}} {{end}} // WIth variable and index
```
https://play.golang.org/p/f5KCS0Yp207